### DISCUSSION TITLE
Fail to decode NR-RRC messages after installing the RLS-wireshark-dissector

### DISCUSSION THREAD

**Discussion by pychan88** [nextmn/.github#11] [Q&A] [answered]
1) My OS version: Ubuntu Linux 
# lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 22.04.3 LTS
Release:	22.04
Codename:	jammy

2) Wireshark version:
Version 4.0.6 (Git v4.0.6 packaged as 4.0.6-1~exp1~ubuntu22.04.0~ppa1)

3) Cannot decode your example pcap file 

![Screenshot from 2024-01-05 16-10-33](https://github.com/nextmn/.github/assets/155712311/141e0aab-b8c3-4814-af7d-f8e835bdfff9)

4) UERANSIM version: 3.2.6

5) Version of plugin: unknown (how to check it?)

6) Plugin can be installed successfully in Terminal but it cannot be found in Wireshark even after reloading the plugin (Analyze > Reload Lua Plugins):

eecatchan@eecatchan-Virtual-Machine:~$ mkdir -p ${XDG_LIB_HOME:-~/.local/lib}/wireshark/plugins
eecatchan@eecatchan-Virtual-Machine:~$ git -C ${XDG_LIB_HOME:-~/.local/lib}/wireshark/plugins clone https://github.com/louisroyer/RLS-wireshark-dissector
Cloning into 'RLS-wireshark-dissector'...
remote: Enumerating objects: 120, done.
remote: Counting objects: 100% (60/60), done.
remote: Compressing objects: 100% (56/56), done.
remote: Total 120 (delta 30), reused 8 (delta 3), pack-reused 60
Receiving objects: 100% (120/120), 58.94 KiB | 1.25 MiB/s, done.
Resolving deltas: 100% (54/54), done.

![Screenshot from 2024-01-05 16-06-32](https://github.com/nextmn/.github/assets/155712311/c9e956bb-1623-494e-b456-b4be39ffd6ea)
![Screenshot from 2024-01-05 16-07-57](https://github.com/nextmn/.github/assets/155712311/2c356528-f368-4e69-8c65-8e97167c4236)

**Comment by louisroyer** [comment, marked as answer]
You have installed the module in user `eecatchan` directory, but you are running Wireshark as `root` (probably because you started it with `sudo`).  Therefore the module cannot be loaded. Please either:
1. Install the plugin as root (i.e. in `/root/.local/lib/wireshark/plugins`), or
2. Run Wireshark as user  `eecatchan`. If you do this, you may want to [configure Wireshark to allow capturing on network interfaces for this user](https://gitlab.com/wireshark/wireshark/-/raw/master/packaging/debian/README.Debian).

**Comment by pychan88** [comment]
Thanks a lot for your help. I can have the RRC messages decoded in Wireshark now with your example capture file. However, it doesn't work with my own capture. Our network is not using UERANSIM gNB simulator but is using OAI. Could this dissector be used to decode the NR-RRC messages in real-time or offline with Wireshark? Thanks again.


________________________________

**Comment by louisroyer** [comment]
My dissector can be used in real-time, but only with UERANSIM's protocol, not OAI's one. The NR-RRC support is builtin in Wireshark and this plugin only deals with decoding the encapsulation layer; I don't know how NR-RRC is encapsulated in OAI softmodem (I guess this is done in https://gitlab.eurecom.fr/oai/openairinterface5g/-/tree/develop/openair1/SIMULATION?ref_type=heads but I can't find a documented protocol).
